### PR TITLE
Teamcenter Extension and Teamcenter Extension Sample App Documentation

### PR DESCRIPTION
**Pull request by wanchoorohan**
This includes documentation for
1. a new Mendix Extension called "Teamcenter Extension"
2. Teamcenter Extension Sample App

Both products will be uploaded to Marketplace this week. Once it is published, we'll later update the documentation with relevant links to Marketplace.

### CODE REVIEW

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 1]
```diff
@@ -8,139 +8,155 @@ tags: ["Teamcenter", "Integration", "PLM", "SOA connector", "low-code"]
...
-* Comparing BOMs from different systems (use Active Integration Gateway)
-* For BOM authoring (use Active Workspace)
-* For creating BOM configurations (use Active Workspace)
+This feature is designed specifically for generating microflows and domain models to retrieve and display simple BOM structures (unconfigured or configured). For other scenarios, consider alternative solutions. See the table below:
 
-## 3 Landing Page {#homepage}
-Open the Teamcenter Extension by clicking on menu option View -> Teamcenter inside Mendix Studio Pro. This will open the landing page. You will see two tabs: Menu and Settings. 
+| Scenario                                                     | Suggested Solution                |
+| ------------------------------------------------------------ | --------------------------------- |
+| You work with large or complex BOM structures                | Use an active workspace           |
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md]: Please remove "an" before active workspace and active integration gateway. Also, the first letter should be capitalized i.e. Active Workspace and Active Integration Gateway

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 2]
```diff
@@ -8,139 +8,155 @@ tags: ["Teamcenter", "Integration", "PLM", "SOA connector", "low-code"]
...
+| You have performance concerns                                | Use an active workspace           |
+| You transfer an entire BOM from Teamcenter to another system | Use an active integration gateway |
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md]: I like the table but not a big fan of the word "You". It doesn't sound right to the TC technical audience. I would suggest the following:

Performance concerns
Transfer an entire BOM from Teamcenter to another system
Compare BOM from different system
Author BOMs
Create BOM configurations

**Review by wanchoorohan — COMMENTED** [review]
Added few comments.

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+* [Community Commons](https://marketplace.mendix.com/link/component/170)
+
+### 1.5 Demo App
+
+To see Teamcenter Extension in action, download and play with the Teamcenter Extension Sample App, which is available for free on the Mendix Marketplace.
+
+## 2 Installation
+
+Follow the instructions in [Using Marketplace Content](/appstore/overview/use-content/) to import the Teamcenter Extension into your app.
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md:47]: Should the instructions be written on the marketplace and point to that?

**Review by wanchoorohan — COMMENTED** [review]
Add comment on installation

File: content/en/docs/appstore/modules/teamcenter-extension.md (hunk 2 above)
Comment (Luyao-Zhang-1) [reply content/en/docs/appstore/modules/teamcenter-extension.md]: Thanks. Sounds good. Just one small change: `Performance concerns` -> `Have performance concerns`. This way every line has the same structure.

**Review by wanchoorohan — COMMENTED** [review]
Looks good. We just need to update URLs once the product is on marketplace

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+The Teamcenter Extension offers a list of use cases for which domain models and microflows can be created. After you select a use case, it uses an import mapping approach similar to Mendix [import mapping](/refguide/import-mappings/). Here, the Teamcenter Extension allows you to select data from the business model of your Teamcenter instance. Based on the selected use case, the import mapping and, for some use cases, additional configurations, the Teamcenter Extension generates and updates the domain model for your integration and generates one or more ready-to-use microflows that you can drag and drop into your application logic.
+
+The extension offers the following integration options:
+* Get item revisions
+* Get workspace objects
+* Create item and item revision
+* Update item and item revision
+* Revise item revisions
+* Create BOM window
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md]: One last minute change - Please change the last bullet point from "Create BOM Window" to "Get Structures"

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+#### 3.1.4 Update Item and Item Revision {#updateitem-and-itemrevision}
+
+The `Update Item and Item Revision` action allows you to generate a microflow to update an `Item` with `ItemRevision` or their specializations in Teamcenter and the corresponding domain model. The resulting microflows implements the `Update Properties` action from the Teamcenter Connector. 
+
+#### 3.1.5 Revise Item Revision {#reviseitem-and-itemrevision}
+
+The `Revise Item and Item Revision` action allows you to generate a microflow to revise an `ItemRevision` or its specializations in Teamcenter and the corresponding domain model. The resulting microflow implements the `Revise Object and Update Properties` actions from the Teamcenter Connector. With the `Revise Object` action, a new `ItemRevision` is created, setting the Teamcenter properties that need to be set upon revising. With the `Update Properties` action, the remaining properties are updated in Teamcenter.
+
+#### 3.1.6 Create BOM Window {#create-bom-window}
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md]: Please change the heading from "Create BOM Window" to "Get Structures"

File: content/en/docs/appstore/modules/teamcenter-extension.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+The `Create BOM Window` action allows you to generate microflows and corresponding domain model to configure a BOM window and retrieve structure data from Teamcenter. This feature supports the retrieval of structures with the following:
```
Comment (wanchoorohan) [inline content/en/docs/appstore/modules/teamcenter-extension.md]: Here as well

**Review by wanchoorohan — COMMENTED** [review]
Added couple of last minute changes.

**Pull request merged**